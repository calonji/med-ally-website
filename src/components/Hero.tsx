import { type FC } from 'react';
import { motion } from 'framer-motion';
import { BackgroundEffects } from '@/components/ui/background-effects';
import WaitlistDialog from './WaitlistDialog';

const Hero: FC = () => {
  return (
    <section
      id="hero"
      role="banner"
      aria-label="Main hero section"
      className="relative min-h-screen flex items-center justify-center overflow-hidden"
    >
      <BackgroundEffects variant="gradient" />

      {/* Floating Elements */}
      <motion.div
        className="absolute inset-0 pointer-events-none"
        initial={{ opacity: 0 }}
        animate={{ opacity: 1 }}
        transition={{ duration: 0.5 }}
      >
        {/* DNA Helix Animation */}
        <div className="absolute top-1/4 -left-12 w-24 h-24 bg-blue-500/10 rounded-full blur-xl" />
        <div className="absolute bottom-1/4 -right-12 w-24 h-24 bg-purple-500/10 rounded-full blur-xl" />

        {/* Particles */}
        {[...Array(20)].map((_, i) => (
          <motion.div
            key={i}
            className="absolute w-2 h-2 bg-blue-500/20 rounded-full"
            style={{
              top: `${Math.random() * 100}%`,
              left: `${Math.random() * 100}%`,
            }}
            animate={{
              y: [0, -20, 0],
              opacity: [0.2, 1, 0.2],
            }}
            transition={{
              duration: 3 + Math.random() * 2,
              repeat: Infinity,
              delay: Math.random() * 2,
            }}
          />
        ))}
      </motion.div>

      <div className="container mx-auto px-4 relative z-10">
        <div className="text-center max-w-4xl mx-auto">
          <motion.div
            initial={{ opacity: 0, y: 20 }}
            animate={{ opacity: 1, y: 0 }}
            transition={{ duration: 0.5 }}
          >
            <h1 className="text-5xl md:text-6xl font-bold text-gray-900 mb-6">
              Spend More Time with Patients,{' '}
              <span className="text-transparent bg-clip-text bg-gradient-to-r from-blue-600 to-purple-600">
                Not Paperwork
              </span>
            </h1>
          </motion.div>

          <motion.p
            initial={{ opacity: 0, y: 20 }}
            animate={{ opacity: 1, y: 0 }}
            transition={{ duration: 0.5, delay: 0.2 }}
            className="text-xl text-gray-600 mb-8"
          >
            MedAlly is your AI-powered clinical assistant that simplifies documentation, enhances
            diagnostic accuracy, and supports better patient outcomesâ€”all in one easy-to-use
            platform.
          </motion.p>

          <motion.div
            initial={{ opacity: 0, y: 20 }}
            animate={{ opacity: 1, y: 0 }}
            transition={{ duration: 0.5, delay: 0.4 }}
            className="flex justify-center"
          >
            <WaitlistDialog className="text-lg px-8 py-4" />
          </motion.div>
        </div>
      </div>
    </section>
  );
};

export default Hero;
