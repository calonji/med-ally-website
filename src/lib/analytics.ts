export const trackEvent = (eventName: string, properties?: Record<string, unknown>): void => {
  // Integration with your analytics platform
  console.log('Track Event:', eventName, properties);
}; 