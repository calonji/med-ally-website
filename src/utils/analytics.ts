import ReactGA from 'react-ga4';

interface EventParams {
  category: string;
  action: string;
  label?: string;
  value?: number;
  nonInteraction?: boolean;
}

interface PageViewParams {
  path: string;
  title?: string;
}

export interface UserProperties {
  role?: string;
  plan?: string;
  organization?: string;
  timestamp?: string;
  [key: string]: string | number | boolean | undefined;
}

export class Analytics {
  static isInitialized = false;

  static init(measurementId: string) {
    if (!this.isInitialized && measurementId) {
      ReactGA.initialize(measurementId, {
        gaOptions: {
          cookieFlags: 'SameSite=Strict;Secure',
          cookieDomain: window.location.hostname,
          cookieUpdate: true
        }
      });
      this.isInitialized = true;
    }
  }

  static trackPageView({ path, title }: PageViewParams) {
    if (this.isInitialized) {
      ReactGA.send({ 
        hitType: "pageview", 
        page: path,
        title,
        cookieFlags: 'SameSite=Strict;Secure'
      });
    }
  }

  static trackEvent({ category, action, label, value, nonInteraction = false }: EventParams) {
    if (this.isInitialized) {
      ReactGA.event({
        category,
        action,
        label,
        value,
        nonInteraction
      });
    }
  }

  static trackTiming({ category, variable, value, label }: {
    category: string;
    variable: string;
    value: number;
    label?: string;
  }) {
    if (this.isInitialized) {
      ReactGA.send({
        hitType: "timing",
        timingCategory: category,
        timingVar: variable,
        timingValue: value,
        timingLabel: label
      });
    }
  }

  static trackException({ description, fatal = false }: {
    description: string;
    fatal?: boolean;
  }) {
    if (this.isInitialized) {
      ReactGA.event({
        category: 'Error',
        action: 'Exception',
        label: description,
        nonInteraction: !fatal
      });
    }
  }

  static setUserProperties(properties: UserProperties) {
    if (this.isInitialized) {
      ReactGA.set({
        ...properties,
        timestamp: new Date().toISOString()
      });
    }
  }
} 