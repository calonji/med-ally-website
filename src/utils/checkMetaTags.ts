/**
 * Utility function to check meta tags on the current page
 * This can be called from the browser console
 */
export function checkMetaTags() {
    const title = document.title;
    const description = document.querySelector('meta[name="description"]')?.getAttribute('content') || 'Not found';
    const canonical = document.querySelector('link[rel="canonical"]')?.getAttribute('href') || 'Not found';
    const ogTitle = document.querySelector('meta[property="og:title"]')?.getAttribute('content') || 'Not found';
    const ogDescription = document.querySelector('meta[property="og:description"]')?.getAttribute('content') || 'Not found';
    const ogUrl = document.querySelector('meta[property="og:url"]')?.getAttribute('content') || 'Not found';
    const twitterTitle = document.querySelector('meta[name="twitter:title"]')?.getAttribute('content') || 'Not found';
    const twitterDescription = document.querySelector('meta[name="twitter:description"]')?.getAttribute('content') || 'Not found';

    console.log('=== Meta Tags for Current Page ===');
    console.log(`Page URL: ${window.location.href}`);
    console.log(`Title: ${title}`);
    console.log(`Description: ${description}`);
    console.log(`Canonical: ${canonical}`);
    console.log(`OG Title: ${ogTitle}`);
    console.log(`OG Description: ${ogDescription}`);
    console.log(`OG URL: ${ogUrl}`);
    console.log(`Twitter Title: ${twitterTitle}`);
    console.log(`Twitter Description: ${twitterDescription}`);

    return {
        title,
        description,
        canonical,
        ogTitle,
        ogDescription,
        ogUrl,
        twitterTitle,
        twitterDescription
    };
}

// Export a global function that can be called from the browser console
if (typeof window !== 'undefined') {
    (window as any).checkMetaTags = checkMetaTags;
} 