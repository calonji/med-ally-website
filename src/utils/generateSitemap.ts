import { writeFileSync } from 'fs';
import { format } from 'date-fns';

interface SitemapURL {
  path: string;
  changefreq: 'always' | 'hourly' | 'daily' | 'weekly' | 'monthly' | 'yearly' | 'never';
  priority: number;
}

const domain = 'https://medally.ai';

const urls: SitemapURL[] = [
  { path: '', changefreq: 'daily', priority: 1.0 },
  { path: '/features', changefreq: 'weekly', priority: 0.8 },
  { path: '/pricing', changefreq: 'weekly', priority: 0.8 },
  { path: '/about', changefreq: 'monthly', priority: 0.7 },
  { path: '/contact', changefreq: 'monthly', priority: 0.7 },
  { path: '/privacy', changefreq: 'monthly', priority: 0.5 },
  { path: '/terms', changefreq: 'monthly', priority: 0.5 }
];

const generateSitemap = () => {
  const today = format(new Date(), 'yyyy-MM-dd');
  
  const sitemap = `<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
  ${urls.map(({ path, changefreq, priority }) => `
    <url>
      <loc>${domain}${path}</loc>
      <lastmod>${today}</lastmod>
      <changefreq>${changefreq}</changefreq>
      <priority>${priority}</priority>
    </url>
  `).join('')}
</urlset>`;

  writeFileSync('public/sitemap.xml', sitemap);
};

export default generateSitemap; 