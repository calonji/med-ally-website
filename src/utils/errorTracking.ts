import { Analytics } from './analytics';

export const trackError = (error: Error, componentStack?: string) => {
  Analytics.trackException({
    description: `${error.name}: ${error.message}${componentStack ? `\nComponent: ${componentStack}` : ''}`,
    fatal: false
  });
}; 