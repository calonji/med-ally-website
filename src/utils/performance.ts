import { Analytics } from './analytics';

export const trackPerformanceMetrics = () => {
  if (typeof window === 'undefined' || !('performance' in window)) return;

  try {
    const metrics = performance.getEntriesByType('navigation')[0] as PerformanceNavigationTiming;
    
    if (metrics) {
      Analytics.trackTiming({
        category: 'Page Load',
        variable: 'DOM Content Loaded',
        value: metrics.domContentLoadedEventEnd,
      });
      
      Analytics.trackTiming({
        category: 'Page Load',
        variable: 'Load Complete',
        value: metrics.loadEventEnd,
      });
    }
  } catch (error) {
    console.warn('Failed to track performance metrics:', error);
  }
}; 