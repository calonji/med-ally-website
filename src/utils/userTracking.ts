import { Analytics, type UserProperties } from './analytics';

export const trackUserProperties = (properties: Omit<UserProperties, 'timestamp'>) => {
  Analytics.setUserProperties({
    ...properties,
    timestamp: new Date().toISOString()
  });
};

export const setUserProperties = (properties: UserProperties) => {
  Analytics.setUserProperties(properties);
}; 