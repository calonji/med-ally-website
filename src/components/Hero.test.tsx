import { describe, it, expect, beforeEach, vi } from 'vitest';
import { render, screen, fireEvent } from '@/test/utils';
import { mockAnalytics } from '@/test/mocks/analytics';
import Hero from './Hero';

vi.mock('@/test/mocks/framer-motion');

describe('Hero Component', () => {
  beforeEach(() => {
    mockAnalytics.trackEvent.mockClear();
  });

  it('renders headline', () => {
    render(<Hero />);
    expect(screen.getByText(/Spend More Time with Patients/i)).toBeInTheDocument();
  });

  it('handles CTA click', () => {
    const { getByRole } = render(<Hero />);
    const button = getByRole('button', { name: /Get Started Free/i });
    fireEvent.click(button);
    
    expect(mockAnalytics.trackEvent).toHaveBeenCalledWith({
      category: 'CTA',
      action: 'Click',
      label: 'Hero Get Started'
    });
  });
}); 