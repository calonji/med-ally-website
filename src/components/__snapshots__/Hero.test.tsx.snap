// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero Component > matches snapshot 1`] = `
<div>
  <section
    aria-label="Main hero section"
    class="relative min-h-screen flex items-center justify-center overflow-hidden"
    id="hero"
    role="banner"
  >
    <div
      class="absolute inset-0 overflow-hidden "
    >
      <div
        class="absolute inset-0 bg-gradient-to-br from-blue-50 via-white to-purple-50"
      />
      <div
        class="absolute inset-0 opacity-20"
      />
    </div>
    <div
      class="absolute inset-0 pointer-events-none"
      style="opacity: 0;"
    >
      <div
        class="absolute top-1/4 -left-12 w-24 h-24 bg-blue-500/10 rounded-full blur-xl"
      />
      <div
        class="absolute bottom-1/4 -right-12 w-24 h-24 bg-purple-500/10 rounded-full blur-xl"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 54.87263646418086%; left: 76.94258843473227%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 24.43308004703979%; left: 35.47077136199028%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 34.80413766119024%; left: 64.37453752166542%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 54.11609986577264%; left: 73.64458987273198%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 86.57773617747758%; left: 59.72221342174544%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 29.465642583118033%; left: 11.173471459608763%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 14.572110748851363%; left: 92.28601511624265%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 21.81945092723383%; left: 38.8389421439761%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 9.01130368618861%; left: 15.628958306526641%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 96.0734068000397%; left: 15.010438910648794%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 73.92357129941911%; left: 98.74676791990775%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 79.25215193863315%; left: 77.53761946950495%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 0.0904924695591447%; left: 93.4641682430237%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 91.78738427101183%; left: 11.944916687579132%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 85.18720188800377%; left: 77.04334548991942%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 94.10595530516402%; left: 82.69499157867288%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 34.186599284070574%; left: 61.871301673648446%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 26.316142960247536%; left: 4.160756999314552%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 28.366240924444284%; left: 56.5588139064122%;"
      />
      <div
        class="absolute w-2 h-2 bg-blue-500/20 rounded-full"
        style="top: 4.429021157170743%; left: 10.078978102809376%;"
      />
    </div>
    <div
      class="container mx-auto px-4 relative z-10"
    >
      <div
        class="text-center max-w-4xl mx-auto"
      >
        <div
          style="opacity: 0; transform: translateY(20px);"
        >
          <h1
            class="text-5xl md:text-6xl font-bold text-gray-900 mb-6"
          >
            Spend More Time with Patients,
             
            <span
              class="text-transparent bg-clip-text bg-gradient-to-r from-blue-600 to-purple-600"
            >
              Not Paperwork
            </span>
          </h1>
        </div>
        <p
          class="text-xl text-gray-600 mb-8"
          style="opacity: 0; transform: translateY(20px);"
        >
          MedAlly is your AI-powered clinical assistant that simplifies documentation, enhances diagnostic accuracy, and supports better patient outcomesâ€”all in one easy-to-use platform.
        </p>
        <div
          class="flex flex-col sm:flex-row gap-4 justify-center"
          style="opacity: 0; transform: translateY(20px);"
        >
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary shadow hover:bg-primary/90 h-10 rounded-md px-8 bg-gradient-to-r from-blue-600 to-purple-600 hover:from-blue-700 hover:to-purple-700 text-white"
          >
            Get Started Free
          </button>
          <button
            class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border-input bg-background shadow-sm hover:text-accent-foreground h-10 rounded-md px-8 border-2 hover:bg-gray-50"
          >
            Watch Demo
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
